File: tests/rosettacode/caesar.py
# Caesar cipher with shift 13
def caesar_encrypt(s, shift):
    result = []
    for c in s:
        if c.isupper():
            result.append(chr((ord(c) - ord('A') + shift) % 26 + ord('A')))
        elif c.islower():
            result.append(chr((ord(c) - ord('a') + shift) % 26 + ord('a')))
        else:
            result.append(c)
    return "".join(result)

def caesar_decrypt(s, shift):
    return caesar_encrypt(s, 26 - shift)

text = "The quick brown fox jumps over the lazy dog"
shift = 13
encrypted = caesar_encrypt(text, shift)
decrypted = caesar_decrypt(encrypted, shift)
print(f"Original:  {text}")
print(f"Encrypted: {encrypted}")
print(f"Decrypted: {decrypted}")

File: tests/rosettacode/caesar.go
package main

import "fmt"

func caesarEncrypt(s string, shift int) string {
	result := make([]byte, len(s))
	for i := 0; i < len(s); i++ {
		c := s[i]
		switch {
		case c >= 'A' && c <= 'Z':
			result[i] = byte((int(c-'A')+shift)%26) + 'A'
		case c >= 'a' && c <= 'z':
			result[i] = byte((int(c-'a')+shift)%26) + 'a'
		default:
			result[i] = c
		}
	}
	return string(result)
}

func caesarDecrypt(s string, shift int) string {
	return caesarEncrypt(s, 26-shift)
}

func main() {
	text := "The quick brown fox jumps over the lazy dog"
	shift := 13
	encrypted := caesarEncrypt(text, shift)
	decrypted := caesarDecrypt(encrypted, shift)
	fmt.Printf("Original:  %s\n", text)
	fmt.Printf("Encrypted: %s\n", encrypted)
	fmt.Printf("Decrypted: %s\n", decrypted)
}
